pallets/collator-allowlist/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+use frame_support::{dispatch::DispatchResult, traits::ValidatorRegistration};
+
+use frame_system::ensure_root;
+
+pub use pallet::*;
+
+pub mod weights;
+pub use weights::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub (super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Type representing the weight of this pallet
+		type WeightInfo: WeightInfo;
+
+		/// The Validator Id type
+		type ValidatorId: Member + Parameter + MaybeSerializeDeserialize;
+
+		/// Type representing the underlying validator registration center.
+		/// It offers us the API we need to check whether a collator
+		/// is ready for its duties in the upcoming session.
+		type ValidatorRegistration: ValidatorRegistration<Self::ValidatorId>;
+	}
+
+	// The genesis config type.
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub initial_state: Vec<T::ValidatorId>,
+	}
+
+	// The default value for the genesis config type.
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self {
+				initial_state: vec![],
+			}
+		}
+	}
+
+	// The build of genesis for the pallet.
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			self.initial_state
+				.iter()
+				.for_each(|id| <Allowlist<T>>::insert(id, ()));
+		}
+	}
+
+	/// The collator's allowlist.
+	/// Note: We implement it as a close-enough HashSet: Map<ValidatorId, ()>.
+	#[pallet::storage]
+	pub(super) type Allowlist<T: Config> = StorageMap<_, Blake2_256, T::ValidatorId, ()>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A collator has been added to the allowlist
+		CollatorAdded(T::ValidatorId),
+
+		/// A collator has been removed from the allowlist
+		CollatorRemoved(T::ValidatorId),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The collator has already been added to the allowlist.
+		CollatorAlreadyAllowed,
+
+		/// The collator is not ready yet following to the underlying `T::ValidatorRegistration`
+		CollatorNotReady,
+
+		/// The provided collator was not found in the storage.
+		CollatorNotPresent,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Add the given `collator_id` to the allowlist.
+		/// Fails if
+		///   - `origin` fails the `ensure_root` check
+		///   - `collator_id` did not yet load their keys into the session pallet
+		///   - `collator_id` is already part of the allowlist
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::allow())]
+		pub fn add(origin: OriginFor<T>, collator_id: T::ValidatorId) -> DispatchResult {
```

Comment:
maybe sync `WeightInfo::allow()` and extrinsic `add()` to the same name? Or is there some reasoning behind this?

Reply:
Good catch! The reason for the misalignment was that first we went from `whitelist` to `allowlist` but then having an extrinsic `allowlist` didn't read well so I change that to `add` and forgot to update the weight. Thanks for catching this, it's now in sync 😎